tests/Unit/ParallelAlgorithms/Interpolation/Test_PostInterpolationCallbacks.cpp

```diff
@@ -514,8 +561,58 @@ SPECTRE_TEST_CASE(
   check_file_contents(expected_integral_a, expected_legend_a, "/SurfaceA");
   check_file_contents(expected_integral_b, expected_legend_b, "/SurfaceB");
   check_file_contents(expected_integral_c, expected_legend_c, "/SurfaceC");
+
+  // Check that the Surfaces file contains the correct surface data
+  [&surfaces_file_prefix]() {
+    // Parameters chosen to match SurfaceD choices above
+    constexpr size_t l_max = 10;
+    constexpr size_t m_max = 10;
+    constexpr double sphere_radius = 2.8;
+    constexpr std::array<double, 3> center{{0.0, 0.0, 0.0}};
+    const Strahlkorper<Frame::Inertial> strahlkorper{l_max, m_max,
+                                                     sphere_radius, center};
+    const YlmSpherepack& ylm = strahlkorper.ylm_spherepack();
+    const std::vector<size_t> extents{
+        {ylm.physical_extents()[0], ylm.physical_extents()[1]}};
+    const std::array<DataVector, 2> theta_phi = ylm.theta_phi_points();
+    const DataVector theta = theta_phi[0];
+    const DataVector phi = theta_phi[1];
+    const DataVector sin_theta = sin(theta);
+    const DataVector radius = ylm.spec_to_phys(strahlkorper.coefficients());
+    const std::string grid_name{"SurfaceD"};
+
+    const auto x{radius * sin_theta * cos(phi)};
+    const auto y{radius * sin_theta * sin(phi)};
+    const auto z{radius * cos(theta)};
+    const std::vector<DataVector> tensor_and_coord_data{
+        x, y, z, square(2.0 * x + 3.0 * y + 5.0 * z)};
+    const std::vector<TensorComponent> tensor_components{
+        {grid_name + "/InertialCoordinates_x", tensor_and_coord_data[0]},
+        {grid_name + "/InertialCoordinates_y", tensor_and_coord_data[1]},
+        {grid_name + "/InertialCoordinates_z", tensor_and_coord_data[2]},
+        {grid_name + "/Square", tensor_and_coord_data[3]}};
+
+    const std::vector<Spectral::Basis> bases{
+        2, Spectral::Basis::SphericalHarmonic};
+    const std::vector<Spectral::Quadrature> quadratures{
+        {Spectral::Quadrature::Gauss, Spectral::Quadrature::Equiangular}};
+    const observers::ObservationId observation_id{0., "/SurfaceD.vol"};
+    TestVolumeData_detail::check_volume_data(
+        surfaces_file_prefix + ".h5"s, 4, grid_name, observation_id.hash(),
+        observation_id.value(), tensor_and_coord_data, {grid_name}, {bases},
+        {quadratures}, {extents},
+        {"InertialCoordinates_x"s, "InertialCoordinates_y"s,
+         "InertialCoordinates_z"s, "Square"s},
+        {{0, 1, 2, 3}}, 1.e-2);  // loose tolerance because of low resolution
+                                 // in the volume, which limits interpolation
+                                 // accuracy
+  }();
+
   if (file_system::check_if_file_exists(h5_file_name)) {
```

Comment:
why intermix the remove here with the added test?

Reply:
No reason...I'll just move this